### ISSUE TITLE
Refactor whodata_callback

### ISSUE THREAD

**Issue by Molter73** [wazuh/wazuh#5014]
|Wazuh version|Component|Install type|Platform|
|---|---|---|---|
| 3.13.0 | Syscheckd | Agent | Windows |

## Description
The `whodata_callback` function is responsible for rendering events from Windows Event Logs, parsing and analyzing them. Because it takes on many tasks, it has become quite large, making it hard to maintain and improve. We will undertake a refactor of it in two stages.

### Stage 1
We will break the function down in 3 different pieces:
- Render
- Parse
- Analyze

In this first stage, we will take the render and parse parts out on their own functions.

- Render
https://github.com/wazuh/wazuh/blob/1f4cc0a9e010fc6137dc1ee91fa6980054e2307d/src/syscheckd/win_whodata.c#L533-L547

- Parse
https://github.com/wazuh/wazuh/blob/1f4cc0a9e010fc6137dc1ee91fa6980054e2307d/src/syscheckd/win_whodata.c#L548-L640

### Stage 2
We will rewrite the switch statement of `whodata_callback` in an attempt to make it easier to understand and maintain.

**Comment by Molter73** [comment]
Working branch: [5014-whodata-callback-refactor](/wazuh/wazuh/tree/5014-whodata-callback-refactor)

Changes so far:
- Created `whodata_event_render` function. It is responsible for rendering the received event and returning a buffer with the result (`NULL` if it fails).
- Parsing of the event has been split into the following functions:
  - `whodata_get_event_id`
  - `whodata_get_handle_id`
  - `whodata_get_access_mask`
  - `whodata_event_parse`
- `whodata_callback` now parses only the required fields on each of the received events.
- Removed `wnode` field from `whodata_evt` struct since it was not being used.
- Removed `w_clist` field from `whodata_evt` struct and associated functions since they were not being used.
- Removed `ignore_remove_event` field from `whodata_evt`, it was being set on several places of `win_whodata.c` but it's value was never checked.
- Removed `delete` field from `whodata_evt`, it was being used along with `ignore_remove_event` but did not have an impact on event generation.
- The 4660 events were being captured solely with the purpouse of setting the `delete` field on the event, it does not serve a purpouse anymore in event generation so it was removed from `whodata_callback` and the subscription query.
- Unraveled a nested if on 4663 events.
- Changed how directory events are processed so it does not depend on the event having a set path, this was masking the real logic which depends on permissions used on the AccessMask on the 4663 events.
- Moved linux specific fields from `whodata_evt` into a `#ifndef WIN32` in order to stop them from taking up resources. Rearranged `free_whodata_event` accordingly.
- Fix some formatting/styling problems.

**Comment by Molter73** [comment]
At a later stage, we will need to address the following conditions that allow a whodata event from a non-monitored directory in these lines:
https://github.com/wazuh/wazuh/blob/42bbcb1fb9984d0389944e113b293d0073c7894b/src/syscheckd/win_whodata.c#L697-L707

This is done in order for later events to remove the audit policies when `fim_checker`.

I've made a small change that allows for `whodata_callback` to generate delete alerts when events with ID 4659 are triggered in Windows.

Added, removed and fixed unit tests accordingly to the changes made to the code.
